code/modules/projectiles/gun_system.dm

```diff
@@ -307,6 +307,18 @@
 	var/obj/screen/ammo/A = user.hud_used.ammo
 	A.add_hud(user, src)
 	A.update_hud(user, src)
+	if(src.attachments["muzzle"] != null)
+		if(istype(src.attachments["muzzle"],  /obj/item/attachable/bayonet) || istype(src.attachments["muzzle"],  /obj/item/attachable/bayonetknife))
+			if(CHECK_BITFIELD(flags_gun_features, GUN_CAN_POINTBLANK))
+				//add the bayonetcharge component
+				var/datum/component/bayonetcharge/comp = GetComponent(/datum/component/bayonetcharge)
+				if(!comp)
+					AddComponent(/datum/component/bayonetcharge)
+				else
+					comp.updatevalues()
```

Comment:
This code is terrible A) getcomponent is a no go in 99% of cases
B) src.var usage
C) snowflake checking for bayonet attachment

Reply:
Get component is gone, src.var is gone ( I think), snowflake is now replaced with signals. :)